File: PresentationStateContainers/iOS_MVP/iOS_MVP/ValidatableField.swift
//
//  ValidatableField.swift
//  iOS_MVP
//
//  Created by Petro Korienev on 5/14/20.
//  Copyright © 2020 Petro Korienev. All rights reserved.
//

import Foundation

struct ValidatableField<ErrorType: Swift.Error> {
    var error: ErrorType?
    var text: String
    let startEditing: ValueCallback<String>?
    let changedEditing: ValueCallback<String>?
    let endEditing: ValueCallback<String>?
}

extension ValidatableField: Defaultable {
    static var defaultValue: ValidatableField<ErrorType> {
        return .init(error: nil,
                     text: "",
                     startEditing: nil,
                     changedEditing: nil,
                     endEditing: nil)
    }
}

File: PresentationStateContainers/Android_MVVM/app/src/main/java/com/soxjke/android_mvvm/ui/main/ValidatableField.kt
package com.soxjke.android_mvvm.ui.main

data class ValidatableField<ErrorType: Throwable>(
    var error: ErrorType? = null,
    var text: String = "",
    val startEditing: ((String) -> Unit)? = null,
    val changedEditing: ((String) -> Unit)? = null,
    val endEditing: ((String) -> Unit)? = null
)